gum/backend-posix/gumexceptor-posix.c

```diff
@@ -378,6 +396,236 @@ gum_exceptor_backend_on_signal (int sig,
   gum_exceptor_backend_detach_handler (self, sig);
 }
 
+static GumMemoryOperation
+gum_exceptor_check_mem_access (gpointer address, 
+                               GumCpuContext * cpu)
+{
+  cs_insn * insn;
+  GumMemoryOperation op;
+
+  insn = disassemble_instruction_at (address, cpu);
+  if (insn == NULL)
+    return GUM_MEMOP_READ;
+
+#if defined (HAVE_I386)
+  op = gum_exceptor_x86_check_mem_operation (insn);
+#elif defined (HAVE_ARM64)
+  op = gum_exceptor_arm64_check_mem_operation (insn);
+#elif defined (HAVE_ARM)
+  op = gum_exceptor_arm_check_mem_operation (insn);
+#else
+  return GUM_MEMOP_READ;
+#endif
+
+  cs_free (insn, 1);
+  return op;
+}
+
+static cs_insn *
+disassemble_instruction_at (gconstpointer address, 
+                            GumCpuContext * cpu)
+{
+  cs_err err;
+  csh capstone;
+  cs_insn * insn = NULL;
```

Comment:
The variable holding the return value should be first. (Others should be chronologically sorted.)